gprofiler/profilers/perf.py

```diff
@@ -77,11 +81,20 @@ def _get_perf_cmd(self) -> List[str]:
             str(self._mmap_sizes[self._type]),
         ] + self._extra_args
 
+    def check_if_restart(self) -> None:
+        if (
+            time.monotonic() - self.start_time >= self._restart_after_s
+            and Process(self._process.pid).memory_info().rss >= self._perf_memory_usage_treshhold  # type: ignore
```

Comment:
Is `rss` enough? As far as I understand, some of the data that perf uses is actually "shared". I'm not sure if it will account in `rss` of psutil.

You can test that with a simple python program that reads many files from e.g `/usr` (after all - this is what perf does, it reads tons of files to extract symbols) and see if it causes its `rss` to increase.

Reply:
Ok, I will check that

Reply:
Added debug prints to check rss and shared values:
```
Type: fp
Rss memory: 4829184
Shared memory: 4186112
Type: dwarf
Rss memory: 8355840
Shared memory: 7704576
Type: fp
Rss memory: 5242880
Shared memory: 4546560
Type: dwarf
Rss memory: 11493376
Shared memory: 10788864
```

> RSS does not include memory that is swapped out. It does include memory from shared libraries as long as the pages from those libraries are actually in memory. It does include all stack and heap memory.

Checking if RSS exceeds threshold should be enough